app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -249,4 +303,73 @@ class DrawActivity : AppCompatActivity() {
             }
         }
     }
+
+    /**
+     * This function is used to setup all the behavior for the bottomSheet.
+     * It sets all of the views in the bottom sheet to set a layout to be visible and expands the sheet.
+     * after making the layout visible, it will animate in via a fade in.
+     * If any button is clicked afterwards, all layouts have animations cancelled.
+     */
+    private fun setupBottomSheet(
+        context: Context,
+        bottomSheet: LinearLayout,
+        viewList: List<View>,
+        layoutList: List<View>
+    ) {
+        behavior = BottomSheetBehavior.from(bottomSheet)
+        viewList.forEach {
+            it.setOnClickListener { view ->
+                for (i in viewList.indices) {
+                    viewList[i].background = null
+                    layoutList[i].apply {
+                        isVisible = false
+                        clearAnimation()
+                    }
+                }
+                when (view.id) {
+                    R.id.pickerButton -> {
+                        viewModel.toolIconClicked(ToolType.GRADIENT)
+                        animateLayout(layoutList[0], context)
+                    }
+                    R.id.brushButton -> {
+                        viewModel.toolIconClicked(ToolType.BRUSH)
+                        animateLayout(layoutList[1], context)
+                    }
+                    R.id.moveButton -> {
+                        viewModel.toolIconClicked(ToolType.MOVE)
+                        animateLayout(layoutList[2], context)
+                    }
+                    R.id.resizeButton -> {
+                        viewModel.toolIconClicked(ToolType.RESIZE)
+                        animateLayout(layoutList[3], context)
+                    }
+                    R.id.filterButton -> {
+                        viewModel.toolIconClicked(ToolType.FILTER)
+                        animateLayout(layoutList[4], context)
+                    }
+                    R.id.layerButton -> {
+                        viewModel.toolIconClicked(ToolType.LAYERS)
+                        animateLayout(layoutList[5], context)
+                    }
+                }
+                view.background = AppCompatResources.getDrawable(
+                    this@DrawActivity,
+                    R.drawable.selected_button_background
+                )
+                behavior.state = BottomSheetBehavior.STATE_EXPANDED
+            }
+        }
+    }
+
+    private fun animateLayout(view: View, context: Context) {
```

Comment:
Make an extension function in another file for this

Reply:
made an Animation Utils file for this